build/dyn_shim_detection.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import subprocess
+import xml.etree.ElementTree as ET
+
+_log = logging.getLogger("dyn-shim-detection")
+show_version_info = project.getProperty("dyn.shim.trace")
+_log.setLevel(logging.DEBUG if show_version_info else logging.INFO)
+# Same as shimplify.py
+ch = logging.StreamHandler()
+ch.setFormatter(logging.Formatter('%(name)s - %(levelname)s - %(message)s'))
+_log.addHandler(ch)
+
+multi_module_project_dir = project.getProperty("maven.multiModuleProjectDirectory")
+expression = project.getProperty("expression")
+
+def __get_buildver(pom, expression):
```

Comment:
Nit: curious why the use of double-underscores here?  Normally only single underscores are used for internal members, like was done for `_log`.